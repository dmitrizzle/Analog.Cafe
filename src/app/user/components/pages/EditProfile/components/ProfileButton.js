import { ButtonStyles } from "@roast-cms/react-button-beans"
import styled from "styled-components"

import { styles } from "../../../../../core/components/controls/Card/components/CardButton"

export default styled.input`
  ${ButtonStyles} ${styles} border: 0;
  user-select: text;
  outline: 0;
  width: 100%;
  cursor: text;
  &:active,
  &:focus,
  &.active {
    background: ${props => props.theme.color.background()} !important;
    color: ${props => props.theme.color.foreground()} !important;
    box-shadow: 0 1px 1px rgba(44, 44, 44, 0.25),
      0 0 0 1px rgba(44, 44, 44, 0.125) !important;
  }
`
