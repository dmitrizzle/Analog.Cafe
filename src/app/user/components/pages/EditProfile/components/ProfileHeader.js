import styled from "styled-components"

import HeaderSmall from "../../../../../core/components/vignettes/HeaderSmall"

export default styled(HeaderSmall)`
  padding: 0;
  input {
    padding: ${props => props.theme.size.block.spacing / 4}em
      ${props => props.theme.size.block.spacing / 2}em;
  }
`
