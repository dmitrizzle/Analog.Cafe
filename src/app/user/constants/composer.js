// default title, subtitle and content ("state") for Composer when blank
export const DEFAULT_COMPOSER_HEADER_STATE = { title: "", subtitle: "" }
