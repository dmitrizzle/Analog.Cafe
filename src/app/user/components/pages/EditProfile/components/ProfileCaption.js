import styled from "styled-components"

import { styles as CaptionStyles } from "../../../../../core/components/vignettes/Caption"
import { styles as CardCaptionStyles } from "../../../../../core/components/controls/Card/components/CardCaption"
import TextareaWithHighlights from "../../../forms/TextInput/components/TextareaWithHighlights"

export default styled(TextareaWithHighlights)`
  ${CaptionStyles} ${CardCaptionStyles} font-style: italic;
`
