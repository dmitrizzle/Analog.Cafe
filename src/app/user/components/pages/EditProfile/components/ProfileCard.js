import React from "react"

import { CardIntegratedForColumns } from "../../../../../core/components/controls/ArticleActions/components/Options"
import { INPUT_SUMMARY_LIMIT } from "../../../../constants/rules-submission"
import { makeFroth } from "../../../../../utils"
import Placeholder from "../../../../../core/components/vignettes/Picture/components/Placeholder"
import ProfileButton from "./ProfileButton"
import ProfileCaption from "./ProfileCaption"

export default props => {
  return (
    <React.Fragment>
      <CardIntegratedForColumns>
        <figure>
          <Placeholder frothId={props.image}>
            <img
              src={makeFroth({ src: props.image, size: "s" }).src}
              alt="Profile avatar"
              onClick={() => props.changeImage()}
              style={{ cursor: "pointer" }}
            />
          </Placeholder>
        </figure>
      </CardIntegratedForColumns>
      <CardIntegratedForColumns>
        <figcaption>
          <ProfileCaption
            maxLength={INPUT_SUMMARY_LIMIT}
            placeholder="Your short author bio goes here."
            onChange={event => props.changeText(event)}
            value={props.text || ""}
            warning={props.warningText}
          />
        </figcaption>
        <ProfileButton
          value={props.buttonText || ""}
          branded={props.buttonText}
          placeholder="Website"
          maxLength={INPUT_SUMMARY_LIMIT}
          onChange={event => props.changeButton(event)}
          onFocus={props.focusButton}
          onBlur={props.blurButton}
        />
      </CardIntegratedForColumns>
    </React.Fragment>
  )
}
