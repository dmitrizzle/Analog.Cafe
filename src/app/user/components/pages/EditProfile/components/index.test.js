import "jest-styled-components"

import { ThemeProvider } from "styled-components"
import { shallow } from "enzyme"
import React from "react"

import { APP_THEME } from "../../../../../../constants"
import ProfileButton from "./ProfileButton"
import ProfileCaption from "./ProfileCaption"
import ProfileCard from "./ProfileCard"
import ProfileHeader from "./ProfileHeader"

it("Render ProfileButton, matches snapshot", () => {
  const element = shallow(
    <ThemeProvider theme={APP_THEME}>
      <ProfileButton />
    </ThemeProvider>
  )
  expect(element).toMatchSnapshot()
})

it("Render ProfileCaption, matches snapshot", () => {
  const element = shallow(
    <ThemeProvider theme={APP_THEME}>
      <ProfileCaption />
    </ThemeProvider>
  )
  expect(element).toMatchSnapshot()
})

it("Render ProfileCard, matches snapshot", () => {
  const element = shallow(<ProfileCard />)
  expect(element).toMatchSnapshot()
})

it("Render ProfileHeader, matches snapshot", () => {
  const element = shallow(
    <ThemeProvider theme={APP_THEME}>
      <ProfileHeader />
    </ThemeProvider>
  )
  expect(element).toMatchSnapshot()
})
