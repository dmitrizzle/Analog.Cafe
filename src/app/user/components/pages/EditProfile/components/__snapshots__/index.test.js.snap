// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProfileButton, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
>
    <styled.input />
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "__styled-components__": [Function],
    "__styled-components__next__": Object {
      "getTheme": [Function],
      "subscribe": [Function],
      "unsubscribe": [Function],
    },
  },
}
`;

exports[`Render ProfileCaption, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
>
    <Styled(Styled(Component)) />
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "__styled-components__": [Function],
    "__styled-components__next__": Object {
      "getTheme": [Function],
      "subscribe": [Function],
      "unsubscribe": [Function],
    },
  },
}
`;

exports[`Render ProfileCard, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Styled(Styled(styled.div))>
          <figure>
                    <styled.div
                              frothId={undefined}
                    >
                              <img
                                        alt="Profile avatar"
                                        onClick={[Function]}
                                        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                                        style={
                                                  Object {
                                                            "cursor": "pointer",
                                                          }
                                        }
                              />
                    </styled.div>
          </figure>
</Styled(Styled(styled.div))>,
        <Styled(Styled(styled.div))>
          <figcaption>
                    <Styled(Styled(Component))
                              maxLength={250}
                              onChange={[Function]}
                              placeholder="Please introduce yourself in 30 words or less."
                              value=""
                              warning={undefined}
                    />
          </figcaption>
          <styled.input
                    maxLength={250}
                    onBlur={undefined}
                    onChange={[Function]}
                    onFocus={undefined}
                    placeholder="Add Your Link Here"
                    value=""
          />
</Styled(Styled(styled.div))>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <figure>
            <styled.div
                        frothId={undefined}
            >
                        <img
                                    alt="Profile avatar"
                                    onClick={[Function]}
                                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                                    style={
                                                Object {
                                                            "cursor": "pointer",
                                                          }
                                    }
                        />
            </styled.div>
</figure>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <styled.div
              frothId={undefined}
>
              <img
                            alt="Profile avatar"
                            onClick={[Function]}
                            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                            style={
                                          Object {
                                                        "cursor": "pointer",
                                                      }
                            }
              />
</styled.div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <img
                alt="Profile avatar"
                onClick={[Function]}
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style={
                                Object {
                                                "cursor": "pointer",
                                              }
                }
/>,
              "frothId": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Profile avatar",
                "onClick": [Function],
                "src": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
                "style": Object {
                  "cursor": "pointer",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": [Function],
          },
          "type": "figure",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <figcaption>
              <Styled(Styled(Component))
                            maxLength={250}
                            onChange={[Function]}
                            placeholder="Please introduce yourself in 30 words or less."
                            value=""
                            warning={undefined}
              />
</figcaption>,
            <styled.input
              maxLength={250}
              onBlur={undefined}
              onChange={[Function]}
              onFocus={undefined}
              placeholder="Add Your Link Here"
              value=""
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Styled(Styled(Component))
                maxLength={250}
                onChange={[Function]}
                placeholder="Please introduce yourself in 30 words or less."
                value=""
                warning={undefined}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "maxLength": 250,
                "onChange": [Function],
                "placeholder": "Please introduce yourself in 30 words or less.",
                "value": "",
                "warning": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "figcaption",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "maxLength": 250,
              "onBlur": undefined,
              "onChange": [Function],
              "onFocus": undefined,
              "placeholder": "Add Your Link Here",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Styled(Styled(styled.div))>
            <figure>
                        <styled.div
                                    frothId={undefined}
                        >
                                    <img
                                                alt="Profile avatar"
                                                onClick={[Function]}
                                                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                                                style={
                                                            Object {
                                                                        "cursor": "pointer",
                                                                      }
                                                }
                                    />
                        </styled.div>
            </figure>
</Styled(Styled(styled.div))>,
          <Styled(Styled(styled.div))>
            <figcaption>
                        <Styled(Styled(Component))
                                    maxLength={250}
                                    onChange={[Function]}
                                    placeholder="Please introduce yourself in 30 words or less."
                                    value=""
                                    warning={undefined}
                        />
            </figcaption>
            <styled.input
                        maxLength={250}
                        onBlur={undefined}
                        onChange={[Function]}
                        onFocus={undefined}
                        placeholder="Add Your Link Here"
                        value=""
            />
</Styled(Styled(styled.div))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <figure>
              <styled.div
                            frothId={undefined}
              >
                            <img
                                          alt="Profile avatar"
                                          onClick={[Function]}
                                          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                                          style={
                                                        Object {
                                                                      "cursor": "pointer",
                                                                    }
                                          }
                            />
              </styled.div>
</figure>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <styled.div
                frothId={undefined}
>
                <img
                                alt="Profile avatar"
                                onClick={[Function]}
                                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                                style={
                                                Object {
                                                                "cursor": "pointer",
                                                              }
                                }
                />
</styled.div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <img
                  alt="Profile avatar"
                  onClick={[Function]}
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style={
                                    Object {
                                                      "cursor": "pointer",
                                                    }
                  }
/>,
                "frothId": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Profile avatar",
                  "onClick": [Function],
                  "src": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
                  "style": Object {
                    "cursor": "pointer",
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": [Function],
            },
            "type": "figure",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <figcaption>
                <Styled(Styled(Component))
                                maxLength={250}
                                onChange={[Function]}
                                placeholder="Please introduce yourself in 30 words or less."
                                value=""
                                warning={undefined}
                />
</figcaption>,
              <styled.input
                maxLength={250}
                onBlur={undefined}
                onChange={[Function]}
                onFocus={undefined}
                placeholder="Add Your Link Here"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Styled(Styled(Component))
                  maxLength={250}
                  onChange={[Function]}
                  placeholder="Please introduce yourself in 30 words or less."
                  value=""
                  warning={undefined}
/>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "maxLength": 250,
                  "onChange": [Function],
                  "placeholder": "Please introduce yourself in 30 words or less.",
                  "value": "",
                  "warning": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "figcaption",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "maxLength": 250,
                "onBlur": undefined,
                "onChange": [Function],
                "onFocus": undefined,
                "placeholder": "Add Your Link Here",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Render ProfileHeader, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ThemeProvider
    theme={
        Object {
            "color": Object {
              "background": [Function],
              "brand": [Function],
              "foreground": [Function],
              "highlight": [Function],
            },
            "effects": Object {
              "borderRadius": Object {
                "med": 0.5,
                "small": [Function],
              },
            },
            "elements": Object {
              "thickBorder": [Function],
            },
            "layer": Object {
              "card": 30,
              "nav": 20,
              "overlay": 40,
              "tuck": -1,
              "up": 10,
            },
            "opacity": Object {
              "half": 0.5,
              "least": 0.125,
              "most": 0.85,
            },
            "size": Object {
              "block": Object {
                "border": 8,
                "column": Object {
                  "l": 820,
                  "m": 750,
                },
                "padding": 1.5,
                "spacing": 1,
              },
              "breakpoint": Object {
                "exact": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "max": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "min": Object {
                  "l": [Function],
                  "m": [Function],
                  "s": [Function],
                  "xl": [Function],
                  "xs": [Function],
                  "xxl": [Function],
                },
                "stops": Object {
                  "max": 1600,
                  "min": 320,
                },
              },
              "font": Object {
                "auto": Array [
                  "
                ",
                  "
        		@media (min-width: ",
                  "32.5625",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "32.5",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (max-width: ",
                  "20",
                  "em) {
        			",
                  "font-size: 	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  " ",
                  "
        		@media (min-width: ",
                  "100.0625",
                  "em) {
        			",
                  "font-size:	",
                  [Function],
                  "px;",
                  "
        		}
        	",
                  ";
              ",
                ],
                "l": 23,
                "m": 20,
                "make": Object {
                  "larger": 3,
                  "normal": 1,
                  "smaller": 0.85,
                  "tiny": 0.65,
                },
                "s": 18,
                "xs": 17,
              },
            },
            "typography": Object {
              "text": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Lora', Georgia, serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                ",
                ],
                "font": "'Lora', Georgia, serif",
                "lineHeight": 1.75,
              },
              "title": Object {
                "auto": Array [
                  "
                  color: ",
                  [Function],
                  ";
                  font-family: ",
                  "'Exo 2', Arial, sans-serif",
                  ";
                  letter-spacing: 0.025em;
                  line-height: ",
                  [Function],
                  "em;
                  font-weight: ",
                  "600",
                  ";
                  margin: 0;
                ",
                ],
                "font": "'Exo 2', Arial, sans-serif",
                "lineHeight": 1.15,
              },
            },
          }
    }
>
    <Styled(styled.header) />
</ThemeProvider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): Object {
    "__styled-components__": [Function],
    "__styled-components__next__": Object {
      "getTheme": [Function],
      "subscribe": [Function],
      "unsubscribe": [Function],
    },
  },
}
`;
